apps/api/src/app/environments-v1/usecases/output-renderers/email-schema-expander.usecase.ts

```diff
@@ -2,76 +2,146 @@
 import { TipTapNode } from '@novu/shared';
 import { ExpendEmailEditorSchemaCommand } from './expend-email-editor-schema-command';
 
-// Rename the class to ExpendEmailEditorSchemaUseCase
 export class ExpandEmailEditorSchemaUsecase {
   execute(command: ExpendEmailEditorSchemaCommand): TipTapNode {
-    return this.expendSchema(command.schema);
-  }
-
-  private expendSchema(schema: TipTapNode): TipTapNode {
-    // todo: try to avoid !
-    const content = schema.content!.map(this.processNodeRecursive.bind(this)).filter(Boolean) as TipTapNode[];
+    this.traverseAndAugment(command.schema, undefined);
 
-    return { ...schema, content };
+    return command.schema;
   }
 
-  private processItemNode(node: TipTapNode, item: any): TipTapNode {
-    if (node.type === 'text' && typeof node.text === 'string') {
-      const regex = /{#item\.(\w+)#}/g;
-      node.text = node.text.replace(regex, (_, key: string) => {
-        const propertyName = key;
-
-        return item[propertyName] !== undefined ? item[propertyName] : _;
+  private traverseAndAugment(node: TipTapNode, parentNode?: TipTapNode) {
```

Comment:
Nitpicky comment to improve readability and self-documentation

```suggestion
  private processShowAndForControls(node: TipTapNode, parentNode?: TipTapNode) {
```